src/main/java/org/apache/commons/imaging/formats/tiff/write/TiffOutputField.java

```diff
@@ -66,6 +66,10 @@ public boolean bytesEqual(final byte[] data) {
         return Arrays.equals(bytes, data);
     }
 
+    public boolean bytesEqual(final TiffOutputField other) {
+        return Arrays.equals(bytes, other.bytes);
+    }
```

Comment:
The original test did not compare the bytes. This is needed to really detect data truncation.